core/infra/config/peer_service_with_json.cpp

```diff
@@ -16,8 +16,23 @@ limitations under the License.
 
 #include "peer_service_with_json.hpp"
 
+
 namespace config {
-    PeerServiceConfig::PeerServiceConfig() {}
+
+    std::set< peer::Node > PeerServiceConfig::peerList;
+    PeerServiceConfig::PeerServiceConfig() {
+        if( peerList.empty() ) {
+            if (auto config = openConfig(getConfigName())) {
+                for (const auto& peer : (*config)["group"].get<std::vector<json>>()){
+                    peerList.insert( peer::Node(
```

Comment:
Tempolary `peer::Node` object will be created here though it might be not so much cost.
`emplace()` could be used instead of `insert()`.